src/app/modules/dealers/services/dealers.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, Inject } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { Dealers } from '../models/dealers.entity';
+
+@Component()
+export class DealersService {
+    constructor(@Inject('DealersRepositoryToken') private readonly dealersRepository: Repository<Dealers>) {
+    }
+
+   mysql_real_escape_string(str: string) {
+        return str.replace(/[\0\x08\x09\x1a\n\r"'\\\%]/g, (char) => {
+            switch (char) {
+                case '\0':
+                    return '\\0';
+                case '\x08':
+                    return '\\b';
+                case '\x09':
+                    return '\\t';
+                case '\x1a':
+                    return '\\z';
+                case '\n':
+                    return '\\n';
+                case '\r':
+                    return '\\r';
+                case '\"':
+                case `'`:
+                case '\\':
+                case '%':
+                    return '\\' + char;
+            }
+        });
+    }
+
+    async findAll(): Promise<Dealers[]> {
+        return await this.dealersRepository.find();
+    }
+
+    async findAllResidential(): Promise<Dealers[]> {
```

Comment:
Let's pull 'US' into a CountryCode enum. This may be a common thing. or it may be specific to the dealers module.